### ISSUE TITLE
Header Data Problem

### ISSUE THREAD

**Issue by mouthdog** [sathvik87/stickyheader-recyclerview-example#1]
I had to clone your project and when I run the application the header data like "YEAR 1987", "YEAR 1990" alway back to "YEAR 1987". Try to logcat header position it shows header position alway back to 0. This problem will occur when we try to scroll up the new header to the top after new header replaces old header then the header will change back to "YEAR 1987".  How to solve this problem?

**Comment by sathvik87** [comment]
@mouthdog Yep, I fixed the problem, I was looping position variable but checking itemPosition. Since in Kotlin, method parameters are final, I had to store itemPosition in a temp variable position but did not use it to correctly.

**Comment by mouthdog** [comment]
@sathvik87 Thank you so much. Everything looks good now.

**Comment by KelvnPere** [comment]
Since this has been fixed, then the issue should be closed. @sathvik87
